include/flatbuffers/flatbuffers.h

```diff
@@ -492,6 +492,12 @@ template<typename T, uint16_t length> class Array {
   const T *data() const { return reinterpret_cast<const T *>(Data()); }
   T *data() { return reinterpret_cast<T *>(Data()); }
 
+  // Copy data from flatbuffers::span with endian conversion.
+  void LoadFrom(flatbuffers::span<const T, length> src) {
+    // FLATBUFFERS_LITTLEENDIAN can be used for optimization after benchmarking.
```

Comment:
I'd hope that most uses of arrays are not that big that it makes a difference. Though maybe it does if you have a vector of struct that contains an array..

Reply:
I added an agressive type-based strategy for data copying.
- type of Array elements is trivially copyable,
- data source used for initialization in generated `ctor` never overlaps with memory of Array.

So better to use `memcpy` whenever it possible. The `memcpy` has great advantage - it erase memory aliasing.
As far as a size of copying memory is known at complie time a compiler can select the best strategy. 
This approach can be used to eliminate performance problems like #6154.